### DISCUSSION TITLE
Next.js static website hosted on Cloudfront redirects to Home page on page refresh at other endpoints

### DISCUSSION THREAD

**Discussion by Vaidehi1710** [vercel/next.js#46869] [Help]
### Summary

The Next.js static website is hosted on **S3 and Cloudfront** and the issue faced is with the routing and refreshing from endpoints other than the home page. 

Directory structure: 
```
pages/
     index.tsx
     search/
          advSearch.tsx
          results.tsx

```
Currently, the Next.js config is set to **trailingSlash: true**
 When next build && next export is executed the exported static files look like this.

```
out/
     index.html
     search/
          advSearch/
               index.html
          results/
               index.html
```
So upon visiting URL `MySearchApp.cloudfront.net`, the correct index.html of the home page is rendered. Upon navigating to advSearch and results pages through links on home page, the routing works correctly. But when the site is refreshed from the advSearch or results page (or visiting URLs MySearchApp.cloudfront.net/results or MySearchApp.cloudfront.net/advSearch directly) the Home Page index.html is rendered. 
Accessing the files by manually adding index.html (i.e `advSearch/index.html`) to the URL works as expected, but just trying to access `advSearch/` doesn't work.

**Expected Behavior** Upon accessing `MySearchApp.cloudfront.net/advSearch`, it should not render Home Page but should point to advSearch/index.html and render the correct page for the endpoint.
### Additional information

Next.js version 12.3

**Comment by icyJoseph** [comment]
You need to make sure that your S3/Cloudfront settings are not set to routing everything to `index.html` - There's something called default root object. 

Have you had a CRA app deployed before on this domains? 

I think this issue might be coming from S3 though, because in Cloudfront one configures the 403/404 behavior instead.
